### PR TITLE
Rotatable 2D map

### LINKED ISSUE #3897: Enable rotation for 2D
Several users on the forum have asked if it is possible to revert to the old 2D scene mode behavior because they want to be able to rotate the 2D map.

https://groups.google.com/forum/?hl=en#!topic/cesium-dev/CPnsBqMzUVo
https://groups.google.com/d/msg/cesium-dev/bUa1zY8RyKM/GBspqvCPAAAJ

### PR DESCRIPTION
**Pull request by bagnell**
Add a `rotatable2D` option to to `Scene`, `CesiumWidget` and `Viewer` to enable a rotatable map in 2D.

Fixes #3897.

### CODE REVIEW

**Comment by pjcozzi** [comment]
@bagnell we'll test now and ship in 1.22 if nothing jumps up.

**Comment by pjcozzi** [comment]
When this is merged, can you also update the forum post(s) where users were asking about this.

File: Specs/Scene/ScreenSpaceCameraControllerSpec.js [hunk 1]
```diff
@@ -446,6 +447,63 @@ defineSuite([
         expect(camera.frustum.bottom).toEqual(-camera.frustum.top);
     });
 
+    it('rotate counter-clockwise in 2D', function() {
+        setUp2D();
+        scene.rotatable2D = true;
```
Comment (pjcozzi) [inline Specs/Scene/ScreenSpaceCameraControllerSpec.js]: Isn't this property readonly?

Comment (pjcozzi) [inline Specs/Scene/ScreenSpaceCameraControllerSpec.js]: I guess this overwrites the property get function with the boolean?  Perhaps it is more obvious to assign to `_ rotatable2D`?

**Comment by pjcozzi** [comment]
Although it will be trivial, I think this is an important enough example that it should have a Sandcastle example.

File: Specs/Scene/ScreenSpaceCameraControllerSpec.js (hunk 1 above)
Comment (bagnell) [inline Specs/Scene/ScreenSpaceCameraControllerSpec.js]: This isn't a real `Scene` instance. A `MockScene` is created before each test.

**Comment by pjcozzi** [comment]
Otherwise, this is a masterpiece.

File: Specs/Scene/ScreenSpaceCameraControllerSpec.js (hunk 1 above)
Comment (pjcozzi) [inline Specs/Scene/ScreenSpaceCameraControllerSpec.js]: Ah, OK.

**Comment by bagnell** [comment]
I just found two issues after I made the Sandcastle example. The home button doesn't reorient the camera and setting the scene mode on viewer construction shows a blank canvas. I'll let you know when they're fixed.

**Comment by hpinkos** [comment]
Should we add back the ability to set the heading from `Camera.flyTo` and `Camera.setView` in 2D when `rotatable2D: true`?

I would also probably call the flag `rotate2D` instead of `rotatable2D`.

This might be overkill, but maybe have an enum for the 2d mode with the options infinite scroll or rotate.  That way it's obvious you only get one of those options.

**Comment by pjcozzi** [comment]
> This might be overkill, but maybe have an enum for the 2d mode with the options infinite scroll or rotate.

Yeah, I thought about this too.  It's probably the right move.

If the scope here is too big, it is not really needed for 1.22.

**Comment by bagnell** [comment]
@pjcozzi This is ready for another look.

File: CHANGES.md [hunk 2]
```diff
@@ -36,6 +36,7 @@ Change Log
 * Added `CullingVolume.fromBoundingSphere`.
 * Added `debugShowShadowVolume` to `GroundPrimitive`.
 * Fix issue with disappearing tiles on Linux. [#3889](https://github.com/AnalyticalGraphicsInc/cesium/issues/3889)
+* Add a `rotatable2D` option to to `Scene`, `CesiumWidget` and `Viewer` to enable a rotatable map in 2D. [#3897](https://github.com/AnalyticalGraphicsInc/cesium/issues/3897)
```
Comment (pjcozzi) [inline CHANGES.md]: Update this to reflect the recent changes and move to 1.23.

**Comment by pjcozzi** [comment]
That's my only comment.

@hpinkos anything else?

We will merge after the 1.22 release.

**Comment by hpinkos** [comment]
looks good to me! :+1:

**Pull request merged**